src/core/positioning/nmeagnssreceiver.cpp

```diff
@@ -59,3 +61,27 @@ void NmeaGnssReceiver::stateChanged( const QgsGpsInformation &info )
                                                           info.satPrn, info.satInfoComplete, std::numeric_limits<double>::quiet_NaN(), std::numeric_limits<double>::quiet_NaN(),
                                                           0, QStringLiteral( "nmea" ) );
 }
+
+void NmeaGnssReceiver::nmeaSentenceReceived( const QString &substring )
+{
+  if ( mLogFile.isOpen() )
+  {
+    mLogStream << substring << Qt::endl;
+  }
+}
+
+void NmeaGnssReceiver::handleStartLogging()
+{
+  const QStringList appDataDirs = PlatformUtilities::instance()->appDataDirs();
+  if ( !appDataDirs.isEmpty() )
+  {
+    mLogFile.setFileName( QStringLiteral( "%1/nmea-%2.log" ).arg( appDataDirs.at( 0 ), QDateTime::currentDateTime().toString( QStringLiteral( "yyyy-MM-ddThh:mm:ss" ) ) ) );
```

Comment:
Could we put this logs into a dedicated subfolder of appDataDir? Not very important, just to keep it more clean